src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -169,7 +169,7 @@ public async Task StartAsync(CancellationToken cancellationToken)
         private void CheckFileSystem()
         {
             // Shutdown if RunFromZipFailed
-            if (_environment.IsZipDeployment(validate: false))
+            if (_environment.ZipDeploymentAppSettingsExist())
```

Comment:
As noted above, to avoid impacting existing usages like this, it'd be good if the IsRunFromZipOrPackage (or whatever we're renaming it to - maybe we leave the name as it was before?) continued to do the IsValidZipSetting checks that it did before

Reply:
Please note that, before my change, the call was IsZipDeployment(**validate: false**) which meant that the check was doing !string.IsNullOrEmpty, not IsValidZipSetting

Reply:
Answered above, the previous check doesn't fit with the new requirements.